# Code for translating transcript-relative loci to genome-relative loci

This repository contains the code used to translate transcript-relative loci (including the output of TargetFinder (https://github.com/carringtonlab/TargetFinder) into genome-relative loci. If the transcript-relative locations span introns) the output will include multiple genome-relative regions. 

## Script 01: Extracting exon information for loci transcripts (shell)
- Extracts the exon structure of loci transcripts from a genome annotation GFF3 file. 
- Splits the exon structure files into separate GFF3 files for transcripts on the negative and positive strand of the genome. 

**Requirements:**
- A UNIX shell such as BASH

**Input files:**
- A GFF3 file of transcript-relative loci you want to translate (loci_file)
- A genome annotation GFF3 file (annotation_file)

## Script 02: Translating the loci (R)
- Generates a .txt file of genome-relative loci_file

**Requirements:**
- R with tidyr, plyr and dplyr packages installed

**Input files:**
All required input files are generated by script 01. 
- annotation_file.exons.targets.positive.2.3.4.gff3
- annotation_file.exons.targets.negative.2.3.4.gff3
- loci_file.2.3.4.gff3

**Output file**
The output of this pipeline is a .txt file with 5 or more columns. 
1. user-provided locus IDs
2. chromosome
3. genome strand
4. Region 1 end
5. Region 1 start

6. Onwards (if the locus spans multiple exons): Locus end and start positions. 
